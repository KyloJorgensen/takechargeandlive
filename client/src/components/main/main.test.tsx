'use strict';

import * as React from 'react';
import { shallow } from 'enzyme';
import Main from './main';

describe('Render <Main/> component', () => {	
	it('matches the snapshot', () => {
		const wrapper = shallow(<Main />);
		expect(wrapper).toMatchSnapshot();
	});	
	it('should have only one div element', () => {
		const wrapper = shallow(<Main />);
		const div = wrapper.find('div'); 
		expect(div).toHaveProperty('length', 1);
	});
	it('should have a div element with a main class', () => {
		const wrapper = shallow(<Main />);
		const mainClass = wrapper.find('div').hasClass('main'); 
		expect(mainClass).toBeTruthy();
	});
	it('should have only one p element', () => {
		const wrapper = shallow(<Main />);
		const p = wrapper.find('p');
		expect(p).toHaveProperty('length', 1);
	});
	it('should have a p element with text', () => {
		const wrapper = shallow(<Main/>);
		const p = wrapper.find('p');
		const text = p.text();
		expect(text).toEqual('Coming Soon...');
	});
	it('should have only one h1 element', () => {
		const wrapper = shallow(<Main />);
		const h1 = wrapper.find('h1');
		expect(h1).toHaveProperty('length', 1);
	});
	it('should have a h1 element with text', () => {
		const wrapper = shallow(<Main/>);
		const h1 = wrapper.find('h1');
		const text = h1.text();
		expect(text).toEqual('Take Charge and Live');
	});
});