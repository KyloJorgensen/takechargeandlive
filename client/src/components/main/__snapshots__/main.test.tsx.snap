// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render <Main/> component matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Take Charge and Live
        </h1>,
        <p>
          Coming Soon...
        </p>,
      ],
      "className": "main",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Take Charge and Live",
        },
        "ref": null,
        "rendered": "Take Charge and Live",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Coming Soon...",
        },
        "ref": null,
        "rendered": "Coming Soon...",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Take Charge and Live
          </h1>,
          <p>
            Coming Soon...
          </p>,
        ],
        "className": "main",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Take Charge and Live",
          },
          "ref": null,
          "rendered": "Take Charge and Live",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Coming Soon...",
          },
          "ref": null,
          "rendered": "Coming Soon...",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
